docs/02-app/01-building-your-application/01-routing/05-dynamic-routes.mdx

```diff
@@ -21,9 +21,15 @@ Dynamic Segments are passed as the `params` prop to [`layout`](/docs/app/api-ref
 
 For example, a blog could include the following route `app/blog/[slug]/page.js` where `[slug]` is the Dynamic Segment for blog posts.
 
-```tsx filename="app/blog/[slug]/page.js"
+```tsx filename="app/blog/[slug]/page.tsx" switcher
+export default function Page({ params: { slug: string } }) {
+  return <div>My Post: {slug}</div>
+}
+```
+
+```jsx filename="app/blog/[slug]/page.jsx" switcher
 export default function Page({ params }) {
-  return <div>My Post</div>
+  return <div>My Post: {slug}</div>
 }
```

Comment:
This looks like you wrote the same thing twice. Is this a merge conflict gone wrong?

Reply:
No, not the same thing. One is TypeScript and one is JavaScript. This will create a single code block with a language switcher.